routes/api.js

```diff
@@ -7,39 +7,86 @@ const jwt = require('jsonwebtoken');
 const HTTP_STATUS_CODES = require('../http_codes');
 
 router.get('/recipes', function(req, res) {
-    Recipe.find(function(err, recipes) {
-        if (err) {
-            res.json({ code: HTTP_STATUS_CODES.SERVER_ERROR, error: err });
-        }
-        res.send(recipes);
-    });
+    const perPage = 2;
+    const page = Number(req.query.page) || 1;
+    Recipe
+        .find({})
+        .skip((perPage * page) - perPage)
+        .limit(perPage)
+        .exec(function(err, recipes) {
+            if (err) {
+                res.json({ code: HTTP_STATUS_CODES.SERVER_ERROR, error: err });
+            }
+            Recipe.count().exec(function(err, count) {
+                if (err) {
+                    res.json({ code: HTTP_STATUS_CODES.SERVER_ERROR, error: err });
+                }
```

Comment:
should we have return here after handling error?

Reply:
Yes, to stop executing script, when the error occurs. Right?